model/vcs/metrics.yaml

```diff
@@ -0,0 +1,139 @@
+      - ref: vcs.repository.ref.type
+        requirement_level: required
+  - id: metric.vcs.repository.ref.lines_added
+    type: metric
+    metric_name: vcs.repository.ref.lines_added
+    brief: 'The number of lines added in a ref (branch) relative to the default branch (trunk)'
+    instrument: gauge
+    unit: "{line}"
+    stability: experimental
+    attributes:
+      - ref: vcs.repository.url.full
+        requirement_level: required
+      - ref: vcs.repository.ref.name
+        requirement_level: required
+      - ref: vcs.repository.ref.type
+        requirement_level: required
+  - id: metric.vcs.repository.ref.lines_deleted
+    type: metric
+    metric_name: vcs.repository.ref.lines_deleted
+    brief: 'The number of lines deleted in a ref (branch) relative to the default branch (trunk)'
+    instrument: gauge
+    unit: "{line}"
+    stability: experimental
+    attributes:
+      - ref: vcs.repository.url.full
+        requirement_level: required
+      - ref: vcs.repository.ref.name
+        requirement_level: required
+      - ref: vcs.repository.ref.type
+        requirement_level: required
+  - id: metric.vcs.repository.ref.revisions_ahead
+    type: metric
+    metric_name: vcs.repository.ref.revisions_ahead
+    brief: 'The number of revisions (commits) a ref (branch) is ahead of the default branch (trunk)'
+    instrument: gauge
+    unit: "{revision}"
+    stability: experimental
+    attributes:
+      - ref: vcs.repository.url.full
+        requirement_level: required
+      - ref: vcs.repository.ref.name
+        requirement_level: required
+      - ref: vcs.repository.ref.type
+        requirement_level: required
+  - id: metric.vcs.repository.ref.revisions_behind
+    type: metric
+    metric_name: vcs.repository.ref.revisions_behind
+    brief: 'The number of revisions (commits) a ref (branch) is behind the default branch (trunk)'
+    instrument: gauge
+    unit: "{revision}"
+    stability: experimental
+    attributes:
+      - ref: vcs.repository.url.full
+        requirement_level: required
+      - ref: vcs.repository.ref.name
+        requirement_level: required
+      - ref: vcs.repository.ref.type
+        requirement_level: required
+  - id: metric.vcs.repository.ref.time
+    type: metric
+    metric_name: vcs.repository.ref.time
+    brief: 'Time a ref (branch) created from the default branch (trunk) has existed. The `ref.type` attribute will always be `branch`'
+    instrument: gauge
+    unit: "s"
+    stability: experimental
+    attributes:
+      - ref: vcs.repository.url.full
+        requirement_level: required
+      - ref: vcs.repository.ref.name
+        requirement_level: required
+      - ref: vcs.repository.ref.type
+        requirement_level: required
+  - id: metric.vcs.repository.contributor.count
+    type: metric
+    metric_name: vcs.repository.contributor.count
+    brief: 'The number of unique contributors to a repository'
+    instrument: gauge
```

Comment:
shouldn't it be a counter? (additive, monotonically incremental) Or if there are edge cases where it's possible to reduce number of contributors, then updowncounter?


If it's counter (not updowncounter), consider calling it `metric.vcs.repository.contributors`:

https://github.com/open-telemetry/semantic-conventions/blob/ee435c0a039fb2a51f070bd15844b186dde52218/docs/general/metrics.md?plain=1#L133-L139

Reply:
I agree that we should rename this to `metric.vcs.repository.contributors`.
However I think the type should stay a gauge because we record absolute values that are non-additive over different repositories (the same contributor might contribute to many repos).